k9mail/src/main/java/com/fsck/k9/activity/compose/CryptoSettingsDialog.java

```diff
@@ -104,21 +118,41 @@ public void onCryptoStatusSelected(int status) {
         }
         updateView(true);
 
+        OnCryptoModeChangedListener activity = getCryptoModeChangeListenerActivity();
+        if (activity == null) {
+            return;
+        }
+
+        activity.onCryptoModeChanged(currentMode);
+    }
+
+    @Override
+    public void onCheckedChanged(CompoundButton compoundButton, boolean switchState) {
+        OnCryptoModeChangedListener activity = getCryptoModeChangeListenerActivity();
+        if (activity == null) {
+            return;
+        }
+
+        activity.onCryptoCompatChanged(switchState);
+    }
+
+    @Nullable
+    private OnCryptoModeChangedListener getCryptoModeChangeListenerActivity() {
         Activity activity = getActivity();
         if (activity == null) {
             // is this supposed to happen?
```

Comment:
So - is it supposed to happen? Is a null activity an expected scenario? Should it be considered an error case in each of these checks?